### DISCUSSION TITLE
in block coding how to use the count variable in a "repeat nn times" block?

### DISCUSSION THREAD

**Discussion by BertLindeman** [pybricks/support#1909] [Q&A] [answered]
In a "for item in range" block you can take the variable block and use it in the loop.

In a "repeat xx times" block I see no way. The generated variable does not show in e.g. variables.

A picture to show the difference between "repeat" and "for" blocks.
![image](https://github.com/user-attachments/assets/23b8b509-0638-4f81-8de8-3c1569c5b695)

As a circumvention I can use the "for" block and have the variable available.

Just wondering if there is a simple way to get the count variable.

Slightly related: If I start with an empty list, how would I append elements to it in block coding?

Bert

**Comment by laurensvalk** [comment, marked as answer]
> In a "for item in range" block you can take the variable block and use it in the loop. In a "repeat xx times" block I see no way. 

The repeat block is just a simpler variant of the "for item in range" block since not everyone is familiar with the latter. If you want a counter, use the item in range. That's why it is prepopulated with the `range` block already in it.

> Slightly related: If I start with an empty list, how would I append elements to it in block coding?

The same as your existing sample above. Appending to an empty list is no different than appending to a list that already has items:

![image](https://github.com/user-attachments/assets/4baf956d-3323-4551-90cb-45001f24483f)

**Reply by BertLindeman** [reply]
Ah I missed the drop-down.
Problem solved.

Thank you!
